lib/src/result/result.dart

```diff
@@ -36,6 +36,47 @@ sealed class Result<S, F extends Object> {
     }
   }
 
+  /// Similar to [Result()], but works in an asynchronous context.
+  ///
+  /// Creates a context for early return, similar to "Do notation". Works like the Rust "?" operator, which is a
+  /// "Early Return Operator". Here "$" is used as the "Early Return Key". when "$" is used on a type [Err],
+  /// immediately the context that "$" belongs to is returned with that [Err]. e.g.
+  ///
+  /// ```dart
+  ///     FutureResult<int,String> innerFn() async {
+  ///         return Err("message");
+  ///     }
+  ///
+  ///     FutureResult<int, String> innerFn2() async {
+  ///         return Ok(1);
+  ///     }
+  ///
+  ///     FutureResult<int, String> earlyReturn() => Result.earlyAsync(($) async {
+  ///         int y = 2;
+  ///         // the function will stop here since an Err was returned
+  ///         int x = await innerFn()[$];
+  ///         // innerFn2 will not be executed
+  ///         int z = await innerFn2()[$];
+  ///         return Ok(x + y + z);
+  ///       });
+  ///     }
+  ///     expect(await earlyReturn().unwrapErr(), "message");
+  ///```
+  ///
+  /// This should be used at the top level of a function as above. Passing "$" to any other functions, nesting, or
+  /// attempting to bring "$" out of the original scope should be avoided. Also, using the method [catchError] on the futures after
+  /// the `[$]` operator might lead to unexpected behaviour.
+  static Future<Result<S, F>> earlyAsync<S, F extends Object>(
```

Comment:
How do you feel about just `async` rather then `earlyAsync`?

Reply:
It's your repo, your call bro ^^

Reply:
Gotcha, I think I'll go with just `async`. Happy to make the change after merging. I'll tag a release with this and async `Option` as well soon. Good work.